### ISSUE TITLE
uv pip install fails for private index based on Artifactory

### ISSUE THREAD

**Issue by gaborbernat** [astral-sh/uv#1393]
Fails with uv:
```bash
❯ uv pip install --index-url https://magic.com/artifactory/api/pypi/internal-pypi/simple tomli==2.0.1
error: Failed to download: tomli==2.0.1
  Caused by: HTTP status client error (404 Not Found) for url (https://magic.com/artifactory/api/pypi/packages/packages/97/75/10a9ebee3fd790d20926a90a2547f0bf78f371b2f13aa822c759680ca7b9/tomli-2.0.1-py3-none-any.whl#sha256=939de3e7a6161af0c887ef91b7d41a53e7c5a1ca976325f429cb46ea9bc30ecc)
```

Works with pip:

```bash
❯ venv/bin/pip install --index-url https://magic.com/artifactory/api/pypi/internal-pypi/simple tomli==2.0.1
Looking in indexes: https://magic.com/artifactory/api/pypi/internal-pypi/simple
Collecting tomli==2.0.1
  Using cached https://magic.com/artifactory/api/pypi/internal-pypi/packages/packages/97/75/10a9ebee3fd790d20926a90a2547f0bf78f371b2f13aa822c759680ca7b9/tomli-2.0.1-py3-none-any.whl (12 kB)
Installing collected packages: tomli
```

Seems the formed URL is malformed...

**Comment by tafia** [comment]
Same here.
Also would it be possible to get /etc/pip.conf configuration automatically? (for a drop-in replacement?)

**Comment by zanieb** [comment]
Sorry to hear you're encountering it too! See #1404 for `pip.conf`.

**Comment by tafia** [comment]
No worries at all!! Thanks for the great work!

**Comment by zanieb** [comment]
I believe this is an authentication problem (hence the 404) the URL doesn't look malformed to me.

**Comment by gaborbernat** [comment]
I'm not as sure. If it would an authentication issue, it would be 403, not 404. Note the two different URLs uv and pip tries to download:


```diff
+ https://magic.com/artifactory/api/pypi/packages/packages/97/75/10a9ebee3fd790d20926a90a2547f0bf78f371b2f13aa822c759680ca7b9/tomli-2.0.1-py3-none-any.whl
- https://magic.com/artifactory/api/pypi/internal-pypi/packages/packages/97/75/10a9ebee3fd790d20926a90a2547f0bf78f371b2f13aa822c759680ca7b9/tomli-2.0.1-py3-none-any.whl
```

Or if I align it up for you:

```diff
+ https://magic.com/artifactory/api/pypi              /packages/packages/97/75/10a9ebee3fd790d20926a90a2547f0bf78f371b2f13aa822c759680ca7b9/tomli-2.0.1-py3-none-any.whl
- https://magic.com/artifactory/api/pypi/internal-pypi/packages/packages/97/75/10a9ebee3fd790d20926a90a2547f0bf78f371b2f13aa822c759680ca7b9/tomli-2.0.1-py3-none-any.whl
```

**Comment by charliermarsh** [comment]
My guess is that this is the same as https://github.com/astral-sh/uv/pull/1413 since we're missing a segment.

**Comment by zanieb** [comment]
Oh thanks for lining this up for me! I stared at these pretty hard and somehow missed that... I'm unsure if #1413 will resolve this as well, if not it's probably the same kind of bug and we can track down the invalid join.

**Comment by gaborbernat** [comment]
Let me build a release of uv and check 🤔

**Comment by gaborbernat** [comment]
Can confirm it fixes it 👍 

```
❯ target/release/uv pip install tomli
Resolved 1 package in 2.12s
Downloaded 1 package in 562ms
Installed 1 package in 2ms
 + tomli==2.0.1
```

**Comment by tafia** [comment]
Working for me too!
Thanks!
